janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -561,10 +561,15 @@ private void tokenize(SearchParams params, final Mapping mapping, final LuceneCu
                 params.addQuery(new PrefixQuery(term), BooleanClause.Occur.MUST);
             } else throw new IllegalArgumentException("LuceneIndex does not support this predicate with 1 token : " + janusgraphPredicate);
         } else {
-            // at the moment, this is only walked for EQUAL and Text.CONTAINS (String and Text mappings)
+            // at the moment, this is only walked for EQUAL, NOT_EQUAL and Text.CONTAINS (String and Text mappings)
             final BooleanQuery.Builder q = new BooleanQuery.Builder();
+            BooleanClause.Occur occur = BooleanClause.Occur.MUST;
```

Comment:
Assign `MUST` clause in `else` instead of assigning it before `if`. I.e. use the next code:
```
BooleanClause.Occur occur;
if (janusgraphPredicate == Cmp.NOT_EQUAL) {
    q.add(new MatchAllDocsQuery(), BooleanClause.Occur.MUST);
    occur = BooleanClause.Occur.MUST_NOT;
} else {
    occur = BooleanClause.Occur.MUST;
}
```

Reply:
Done